pkg/anonymization/builder_test.go

```diff
@@ -0,0 +1,133 @@
+		dataPolicy       v1alpha1.DataPolicy
+		kubeClient       kubernetes.Interface
+		networkClient    networkv1client.NetworkV1Interface
+		networks         []string
+		runningInCluster bool
+		secretsClient    corev1client.SecretInterface
+	}
+
+	fakeCfgClient := configfake.NewSimpleClientset().ConfigV1()
+	fakeConfigurator := config.NewMockConfigMapConfigurator(&config.InsightsConfiguration{})
+	fakeKubeClient := kubefake.NewSimpleClientset()
+	fakeNetworkClient := networkfake.NewSimpleClientset().NetworkV1()
+	fakeSecretsClient := kubefake.NewSimpleClientset().CoreV1().Secrets("mock")
+
+	testCases := []testCase{
+		{
+			name:    "Basic builder need no settings to create a valid anonymizer",
+			builder: getBuilderInstance(),
+		},
+		{
+			name:            "method 'AddSensitiveValue' sets values on the anonymizer instance",
+			builder:         getBuilderInstance().AddSensitiveValue("mock", "xxxx"),
+			sensitiveValues: map[string]string{"mock": "xxxx"},
+		},
+		{
+			name:         "method 'WithConfigClient' sets the client on the anonymizer instance",
+			builder:      getBuilderInstance().WithConfigClient(fakeCfgClient),
+			configClient: fakeCfgClient,
+		},
+		{
+			name:         "method 'WithConfigurator' sets the configurator on the anonymizer instance",
+			builder:      getBuilderInstance().WithConfigurator(fakeConfigurator),
+			configurator: fakeConfigurator,
+		},
+		{
+			name:       "method 'WithDataPolicy' sets the policy on the anonymizer instance",
+			builder:    getBuilderInstance().WithDataPolicy(v1alpha1.ObfuscateNetworking),
+			dataPolicy: v1alpha1.ObfuscateNetworking,
+		},
+		{
+			name:       "method 'WithKubeClient' sets the client on the anonymizer instance",
+			builder:    getBuilderInstance().WithKubeClient(fakeKubeClient),
+			kubeClient: fakeKubeClient,
+		},
+		{
+			name:          "method 'WithNetworkClient' sets the client on the anonymizer instance",
+			builder:       getBuilderInstance().WithNetworkClient(fakeNetworkClient),
+			networkClient: fakeNetworkClient,
+		},
+		{
+			name:     "method 'WithNetworks' sets the networks on the anonymizer instance",
+			builder:  getBuilderInstance().WithNetworks([]string{"127.0.0.0/8", "192.168.0.0/16"}),
+			networks: []string{"127.0.0.0/8", "192.168.0.0/16"},
+		},
+		{
+			name:             "method 'WithRunningInCluster' sets the value on the anonymizer instance",
+			builder:          getBuilderInstance().WithRunningInCluster(true),
+			runningInCluster: true,
+		},
+		{
+			name:          "method 'WithSecretsClient' sets the client on the anonymizer instance",
+			builder:       getBuilderInstance().WithSecretsClient(fakeSecretsClient),
+			secretsClient: fakeSecretsClient,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// When
+			test, err := tc.builder.Build()
+
+			// Assert
+			assert.NoError(t, err)
+			assert.IsType(t, Anonymizer{}, *test)
+			if tc.sensitiveValues != nil {
```

Comment:
Are these conditions required? I guess when you provide `nil` input, you can also expect `nil` output or?

Reply:
Solved in bc66284efa66957e06efbc3efd2c1f893793085e, thanks for the review